### PR TITLE
Allow `DissimilarityMatrix` to use `np.float32`

### PR DESCRIPTION
**Pull request by wasade**
We observed a Mantel R^2 of 0.9999 between float32 and float64 results with weighted UniFrac when applied to the EMP dataset (~27k samples). This observation arose from a recent GPU adaption of [UniFrac](https://arxiv.org/abs/2005.05826) when testing for a performance difference when computing float32 and float64. We're now producing float32 matrices from UniFrac as it's appreciably faster, however the `DissimilarityMatrix` object at the moment requires double precision. This PR allows `DissimilarityMatrix` to both support float32, and to avoid an implicit copy that would occur if passing in float32 data. 

The implicit copy is important to avoid for large matrices to keep resident memory low.  
 
Please complete the following checklist:

* [x] I have read the guidelines in [CONTRIBUTING.md](https://github.com/biocore/scikit-bio/blob/master/CONTRIBUTING.md).

* [x] I have documented all public-facing changes in [CHANGELOG.md](https://github.com/biocore/scikit-bio/blob/master/CHANGELOG.md).

* [ ] **This pull request includes code, documentation, or other content derived from external source(s).** If this is the case, ensure the external source's license is compatible with scikit-bio's [license](https://github.com/biocore/scikit-bio/blob/master/COPYING.txt). Include the license in the `licenses` directory and add a comment in the code giving proper attribution. Ensure any other requirements set forth by the license and/or author are satisfied. **It is your responsibility to disclose code, documentation, or other content derived from external source(s).** If you have questions about whether something can be included in the project or how to give proper attribution, include those questions in your pull request and a reviewer will assist you.

* [x] **This pull request does not include code, documentation, or other content derived from external source(s).**

**Note:** [REVIEWING.md](https://github.com/biocore/scikit-bio/blob/master/REVIEWING.md) may also be helpful to see some of the things code reviewers will be verifying when reviewing your pull request.

### CODE REVIEW

**Comment by wasade** [comment]
So I think this is good too, cc @gwarmstrong @ElDeveloper

File: .travis.yml [hunk 1]
```diff
@@ -28,7 +28,7 @@ install:
   - conda info --envs
   - python --version
   - pip install -r ci/pip_requirements.txt
-  - pip install .
+  - pip install . --no-deps
```
Comment (ElDeveloper) [inline .travis.yml:31]: What's the reason for this change?

**Review by ElDeveloper — APPROVED** [review]

**Comment by wasade** [comment]
Without it, we risk having the dependency files falling out of sync with
whats actually needed

**Comment by ElDeveloper** [comment]
Looks good to me!

**Comment by wasade** [comment]
Thanks!!
